cmp/form/HoistInput.js

```diff
@@ -7,98 +7,113 @@
- * in that the bound value is only read *within* this control, so that changes to its value do not
- * cause the parent of this control to re-render.
+ * For many fields (e.g. checkbox, select, switchInput, slider) commit always fires at the same time
+ * as the change event. Other fields such as textInput maintain the distinction described above,
+ * but expose a `commitOnChange` prop to force them to eagerly flush their values on every change.
  *
  * HoistInputs support built-in validation when bound to a model enhanced by `@FieldSupport`.
  * When a HoistInput control is linked to a property on the underlying model decorated by `@field`,
  * the model Field will be used to provide validation info and styling to the input component.
+ *
+ * For an even more managed display, consider wrapping HoistInputs in a FormField Component, which
+ * provide out-of-the-box support for labels and validation messages, both read from the Model.
  */
 export class HoistInput extends Component {
 
     static propTypes = {
-
-        /** value of the control */
+        /** Value of the control, if provided directly. */
         value: PT.any,
-        /** handler to fire when value changes, gets passed the new value */
+
+        /** Handler called when value changes - passed the new value. */
         onChange: PT.func,
-        /** handler to fire when value is committed, gets passed the new value */
+
+        /** Handler called when value is committed to backing model - passed the new value. */
         onCommit: PT.func,
-        /** config to commit changes on change instead of blur */
-        commitOnChange: PT.bool,
-        /** model to bind to */
+
+        /** Bound HoistModel instance. */
         model: PT.object,
-        /** name of property in model to bind to */
+
+        /** Property name on bound Model from which to read/write data. */
         field: PT.string,
-        /** is control disabled */
+
+        /** True to disable user interaction. */
         disabled: PT.bool,
-        /** Style block */
+
+        /** Style block. */
         style: PT.object,
-        /** css class name **/
-        className: PT.string
-    };
 
-    static defaultProps = {
-        commitOnChange: false
+        /** CSS class name. **/
+        className: PT.string,
+
+        /** Tab order for focus control, or -1 to skip. If unset, browser layout-based order. **/
+        tabIndex: PT.number
     };
 
     @observable hasFocus;
     @observable internalValue;
 
+    constructor(props) {
+        super(props);
+
+        // Ensure that updates to the external value - sourced from either model or props - are
+        // always flushed to the internal value of this control and reflected in renderValue.
+        this.addReaction({
+            track: () => this.externalValue,
+            run: (externalVal) => {
+                this.setInternalValue(this.toInternal(externalVal));
+            },
+            fireImmediately: true
```

Comment:
Do we need the fireImmediately here?  Does it replace something else that was initializing the internalValue?